hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -323,21 +327,21 @@ public String toString() {
 	 */
 	@FunctionalInterface
 	protected interface FullPathCalculator extends Serializable {
-		String calculateFullPath(NavigablePath parent, String localName, String alias);
+		String calculateFullPath(NavigablePath parent, String localName, @Nullable String alias);
```

Comment:
You were right about this.

```suggestion
		String calculateFullPath(@Nullable NavigablePath parent, String localName, @Nullable String alias);
```